### ISSUE TITLE
import java.util.* ignored as wildcard import

### ISSUE THREAD

**Issue by jbruchanov** [pinterest/ktlint#1614]
## Expected Behavior
Failing ktlintCheck for `wildcard import` rule
```
import java.util.*
```

## Observed Behavior
ktlintCheck task finishes as success

## Steps to Reproduce

.editorconfig
```
[*.{kt,kts}]
disabled_rules=comment-spacing,import-ordering
max_line_length=180
```

ktlint.gradle
```
apply plugin: "org.jlleitschuh.gradle.ktlint"
//https://github.com/jlleitschuh/ktlint-gradle

ktlint {
    version.set(project["ktLint"])
    verbose.set(true)
    android.set(true)
    outputToConsole.set(true)
    outputColorName.set("RED")
}
```

test.kt
```
import java.util.*

fun test() {
    val d = Date()
}

```
run `./gradlew ktlintCheck`
~(doesn't fail even for case of unused imports)~ (doesn't seem to be an issue for 0.47.0)
might be related to https://github.com/pinterest/ktlint/issues/1256

## Your Environment
ktLintPlugin=11.0.0
ktLint=0.45.2 (due to incompatibility plugin vs ktlint 0.46+, this seems to be latest working version)

**Comment by paul-dingemans** [comment]
I am failing to understand your issue. 

You mention at the start of the issue a problem with the wildcard import rule (e.g. `no-wildcard-imports` rule) that it does not report `import java.util.*`. When I use ktlint CLI with version 0.45.2, 0.46.0, 0.46.1 and 0.47.0 the wildcard is reported with all versions.  Would you mind trying to run `ktlint -a -v` on your project with the CLI version (https://pinterest.github.io/ktlint/install/cli/). Please send the log file in case the problem persists.

The part that is confusing is that at the end of the issue you mention that in sample below ktlint even doesn't fail for the case of unused imports:
```
import java.util.*

fun test() {
    val d = Date()
}
```
In this sample `java.util.*` is required for the `Date()` so it is correct that the import is not marked as "unused". But it should report the wildcard import.

**Comment by jbruchanov** [comment]
@paul-dingemans 
`import java.util.*` is just being ignored as wildcard import, I expect to be marked as unwanted as following example for `com.test.*`


running `java -jar ktlint *.kt` with only `Test.kt` in the folder

Example 1
```
import com.test.*
import com.test2.*
import java.util.*

fun test() {
    val d = Date()
}
```
output: (only 2 fails)
```
c:\Temp\ktlint>java -jar ktlint *.kt
c:\Temp\ktlint\Test.kt:1:1: Wildcard import (cannot be auto-corrected) (no-wildcard-imports)
c:\Temp\ktlint\Test.kt:2:1: Wildcard import (cannot be auto-corrected) (no-wildcard-imports)
c:\Temp\ktlint>
```

---------- 
Example 2
```
import java.util.*

fun test() {
    val d = Date()
}
```

output: (unexpected no fail for java.util.*)
```
c:\Temp\ktlint>java -jar ktlint *.kt
c:\Temp\ktlint>
```

**Comment by jbruchanov** [comment]
and the log
[log.txt](https://github.com/pinterest/ktlint/files/9438157/log.txt)

**Comment by jbruchanov** [comment]
just a guess, 
but isn't it this what makes it ignoring ?

https://github.com/pinterest/ktlint/blob/1d9ca64458eb3cee4e21ec8252f22aa58445f4f5/ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/NoWildcardImportsRule.kt#L94

if so, can I disable it somehow ?

**Comment by paul-dingemans** [comment]
That is really a good catch. I can only vaguely remember that I saw this code before. 

You are really close to solve this yourself. Line 94 define the default value for property `ij_kotlin_packages_to_use_import_on_demand`.
```
        public val packagesToUseImportOnDemandProperty: UsesEditorConfigProperties.EditorConfigProperty<List<PatternEntry>> =
            UsesEditorConfigProperties.EditorConfigProperty(
                type = PropertyType(
                    "ij_kotlin_packages_to_use_import_on_demand",
                    "Defines allowed wildcard imports",
                    packagesToUseImportOnDemandPropertyParser,
                ),
                /**
                 * Default IntelliJ IDEA style: Use wildcard imports for packages in "java.util", "kotlin.android.synthetic" and
                 * it's subpackages.
                 *
                 * https://github.com/JetBrains/kotlin/blob/ffdab473e28d0d872136b910eb2e0f4beea2e19c/idea/formatter/src/org/jetbrains/kotlin/idea/core/formatter/KotlinCodeStyleSettings.java#L81-L82
                 */
                defaultValue = parseAllowedWildcardImports("java.util.*,kotlinx.android.synthetic.**"),
                propertyWriter = { it.joinToString(separator = ",") },
            )
```

You can override that property in your `.editorconfig`, but this might result in conflicts with the default formatter of IntelliJ IDEA. But it should :
```
ij_kotlin_packages_to_use_import_on_demand=""
```

**Comment by jbruchanov** [comment]
awesome, it does the trick, so closing it..
thanks...
